contracts/Example3.sol

```diff
@@ -1,14 +1,23 @@
+// SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.4;
 
 contract Example3 {
-    uint256 marginPercentage = 30;
+
+    uint256 constant marginPercentage = 30;
+
     function getOwnerMargin(uint256 amount)
         public
-        view
+        pure
         returns (uint256 amountForSender, uint256 amountForOwner)
     {
-        amountForSender = (amount * (100 - marginPercentage)) / 100;
-        amountForOwner = (amount * marginPercentage) / 100;
+        // uint256 _marginPercentage = marginPercentage;
+        // amountForSender = (amount * (100 - _marginPercentage)) / 100;
+        // amountForOwner = (amount * _marginPercentage) / 100;
+
+        // code below is equivalent; saves 350 gas
+        uint amount_over_100 = amount / 100;
```

Comment:
Is the code actually equivalent? What happens when amount is below 100? Consider that solidity doesn't support floating type

Reply:
HA! I didn't think of that :) good spot!